pandas/core/indexes/multi.py

```diff
@@ -1840,15 +1849,19 @@ def is_lexsorted(self) -> bool:
         ...                            ['bb', 'aa', 'aa', 'bb']]).is_lexsorted()
         False
         """
-        return self.lexsort_depth == self.nlevels
+        return self._lexsort_depth == self.nlevels
 
     @cache_readonly
     def lexsort_depth(self):
-        if self.sortorder is not None:
-            return self.sortorder
-
-        return self._lexsort_depth()
+        warnings.warn(
```

Comment:
you don't need the cache on this one (as its already on _lexsort_depth) and this is now user facing